### PR TITLE
Rubinius popen and test fixes

### PR DESCRIPTION
**Pull request by sodabrew**
The Rubinius work from #551 is rebased here.

### CODE REVIEW

**Comment by sodabrew** [comment]
I've decided that I'm happy with this with respect to rubinius/rubinius#3268.

**Comment by tmm1** [comment]
Diff contains a swp file

**Comment by sodabrew** [comment]
Derp. Removed swp file.

File: tests/em_test_helper.rb [hunk 1]
```diff
@@ -45,6 +45,10 @@ def windows?
     def jruby?
       defined? JRUBY_VERSION
     end
+
+    def rbx?
```
Comment (brixen) [inline tests/em_test_helper.rb]: You may be interested in https://github.com/brixen/redcard

**Pull request merged**